ush/eva/gen_eva_obs_yaml.py

```diff
@@ -17,22 +17,34 @@ def gen_eva_obs_yaml(inputyaml, templateyaml, outputdir):
     except Exception as e:
         logging.error(f'Error occurred when attempting to load: {inputyaml}, error: {e}')
     # get just the observations part of the YAML
-    if 'observers' in jedi_yaml_dict['observations']:
-        jediobs = jedi_yaml_dict['observations']['observers']
+    if 'cost function' in jedi_yaml_dict:
+        # cost function is in marine DA var.yaml
+        if 'observations' in jedi_yaml_dict['cost function']:
+            jediobs = jedi_yaml_dict['cost function']['observations']['observers']
     else:
-        # the unit tests have a different YAML setup
-        jediobs = jedi_yaml_dict['observations']
+        # this is for atm DA var.yaml
+        if 'observers' in jedi_yaml_dict['observations']:
+            jediobs = jedi_yaml_dict['observations']['observers']
+        else:
+            # the unit tests have a different YAML setup
+            jediobs = jedi_yaml_dict['observations']
     # construct a simplified list of obsspaces for EVA
     evaobs = []
     for obsspace in jediobs:
         tmp_os = obsspace['obs space']
+        sim_var = tmp_os['simulated variables'][0]  # to get only the variable, not the list
         tmp_dict = {
             'name': tmp_os['name'],
             'diagfile': tmp_os['obsdataout']['engine']['obsfile'].replace('.nc4', '_0000.nc4'),
             'vars': tmp_os['simulated variables'],
             'channels': tmp_os.get('channels', None),
         }
-        evaobs.append(tmp_dict)
+
+        if variable == sim_var:
```

Comment:
I'm still confused about this, so if the variable you pass is the *first* variable in the simulated variable list, or it is `all`, then you include this dictionary? What if it is the second variable in the list? Seems like the if statement should set the dictionary? or am I not understanding what you are trying to do

Reply:
Got your point. Thanks. It seems better to keep it as original to generate all obs yamls. Modify the script such as `exgdas..._verify.sh` to plot only what we want instead of plotting all.